opm/autodiff/GeoProps.hpp

```diff
@@ -304,6 +291,59 @@ namespace Opm
                              Opm::EclipseStateConstPtr eclState,
                              std::vector<double> &ntg);
 
+        template <class GridType>
+        void computePoreVolume_(const GridType &grid,
+                                Opm::EclipseStateConstPtr eclState)
+        {
+            int numCells = Opm::AutoDiffGrid::numCells(grid);
+            const int* globalCell = Opm::UgGridHelpers::globalCell(grid);
+            EclipseGridConstPtr eclGrid = eclState->getInputGrid();
+            const int nx = eclGrid->getNX();
+            const int ny = eclGrid->getNY();
+
+            // the "raw" pore volume.
+            const std::vector<double>& porvData =
+                eclState->get3DProperties().getDoubleGridProperty("PORV").getData();
+            pvol_.resize(numCells);
+
+            // the "activation number" grid property
+            const std::vector<int>& actnumData =
+                eclState->get3DProperties().getIntGridProperty("ACTNUM").getData();
+
+
+            for (int cellIdx = 0; cellIdx < numCells; ++cellIdx) {
+                const int cellCartIdx = globalCell[cellIdx];
+
+                double cellPoreVolume = porvData[cellCartIdx];
+
+                if (eclGrid->getMinpvMode() == MinpvMode::ModeEnum::OpmFIL) {
+                    // Sum the pore volumes of the cells above which have been deactivated
+                    // because their volume less is less than the MINPV threshold
+                    for (int aboveCellCartIdx = cellCartIdx - nx*ny;
+                         aboveCellCartIdx >= 0;
+                         aboveCellCartIdx -= nx*ny)
+                    {
+                        if (porvData[aboveCellCartIdx] >= eclGrid->getMinpvValue()) {
+                            // stop if we encounter a cell which has a pore volume which is
+                            // at least as large as the minimum one
+                            break;
+                        }
+
+                        const double aboveCellVolume = eclGrid->getCellVolume(aboveCellCartIdx);
+                        if (actnumData[aboveCellCartIdx] == 0 && aboveCellVolume > 1e-6) {
+                            // stop at explicitly disabled cells, but only if their volume is
+                            // greater than 10^-6 m^3
+                            break;
+                        }
```

Comment:
> There is no logic in the gridManager that avoids cells will small volumes.

the purpose of this 'if' is not to avoid these cells, but to make them not stop the pore volume accumulation process. (i.e. if OpmFIL is true and a small cell gets manually disabled, the pore volume of the cells above that manually disabled cell is still summed to the active one below.)

> Will actnum as given by EclState be different than what is given by the grid?

I guess it does not matter: for this condition to trigger, one needs to be in the loop and the loop is only executed for cells above the currently active one which exhibit a pore volume smaller than MINPV.

> Is it possible to check whether aboveCellCartIdx is in the grid?

I suppose that's not (easily) possible: there's only a compressed index -> cartesian index map, not vice-versa. (of course one can create the reverse map, but IMO that would be overkill.)